### PR TITLE
chore: improve loading ux and increase page size to 50

### PR DESCRIPTION
**Pull request by joaoviana**
Closes: #12394

### Description:

Improves loading experience
increases page size to `50`

> [!NOTE]
> on the ticket it says to show the spinner on scroll for next page, but I can't seem to be able to show it. I still think that these changes are better than what we have now.

demo:

https://github.com/user-attachments/assets/9eb68259-e5a3-4624-a467-1d3e9eb4a7a0

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/features/metricsCatalog/components/MetricsTable.tsx [hunk 1]
```diff
@@ -63,6 +71,25 @@ export const MetricsTable = () => {
...
+                'remove-category',
+                'update-catalog-item-icon',
+            ];
+            return Boolean(
+                mutation.options.mutationKey?.some((key) =>
+                    mutationKeys.includes(key as string),
+                ),
+            );
+        },
+    });
```
Comment (joaoviana) [inline packages/frontend/src/features/metricsCatalog/components/MetricsTable.tsx:91]: can be moved to a separate hook? 

wonder how cache keys can be maintained better. constants?

File: packages/frontend/src/features/metricsCatalog/hooks/useCatalogItemIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
+import {
```
Comment (joaoviana) [inline packages/frontend/src/features/metricsCatalog/hooks/useCatalogItemIcon.tsx:1]: moved to separate file

File: packages/frontend/src/features/metricsCatalog/components/MetricsTable.tsx [hunk 3]
```diff
@@ -95,6 +122,11 @@ export const MetricsTable = () => {
         return lastPage.pagination?.totalResults ?? 0;
     }, [data]);
 
+    const showLoadingOverlay = useMemo(
+        () => isFetching && isPreviousData && !isMutating,
```
Comment (almeidabbm) [inline packages/frontend/src/features/metricsCatalog/components/MetricsTable.tsx:126]: Can't we just use `isFetchingNextPage` here?

File: packages/frontend/src/features/metricsCatalog/components/MetricsTable.tsx (hunk 1 above)
Comment (almeidabbm) [reply packages/frontend/src/features/metricsCatalog/components/MetricsTable.tsx:92]: I think having the keys as constants would be a very nice addition here. Don't think we would need to extract this to a separate hook for now, if we end up reusing it then I would do that

**Review by almeidabbm — APPROVED** [review]

File: packages/frontend/src/features/metricsCatalog/components/MetricsTable.tsx (hunk 3 above)
Comment (joaoviana) [reply packages/frontend/src/features/metricsCatalog/components/MetricsTable.tsx:126]: apparently not lol...

**Pull request merged**